src/Compilers/CSharp/Portable/Symbols/TypeWithState.cs

```diff
@@ -15,12 +15,39 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
         public bool HasNullType => Type is null;
         public bool MayBeNull => State == NullableFlowState.MaybeNull;
         public bool IsNotNull => State == NullableFlowState.NotNull;
-        public static TypeWithState ForType(TypeSymbol type) => new TypeWithState(type, type?.CanContainNull() == true ? NullableFlowState.MaybeNull : NullableFlowState.NotNull);
-        public TypeWithState(TypeSymbol type, NullableFlowState state) => (Type, State) = (type, state);
+
+        private static bool CanContainNull(TypeSymbol type)
+        {
+            return type is null || !type.IsValueType || type.IsNullableTypeOrTypeParameter();
```

Comment:
>type is null || !type.IsValueType || type.IsNullableTypeOrTypeParameter() [](start = 19, length = 73)

type?.CanContainNull() != false

Reply:
`TypeSymbolExtensions.CanContainNull()` does not handle type parameters constrained to `Nullable<T>`.

---
In reply to: [271082803](https://github.com/dotnet/roslyn/pull/34630#discussion_r271082803) [](ancestors = 271082803)